iso15118/secc/states/iso15118_2_states.py

```diff
@@ -577,61 +592,120 @@ async def process_message(
+                # to backend.
+                # Call to b64encode returns byte[] - hence the .decode("utf-8")
+                base64_certificate_install_req = base64.b64encode(message_exi).decode(
+                    "utf-8"
+                )
+
+                # The response received below is EXI response in base64 encoded form.
+                # Decoding to EXI happens later just before V2GTP packet is built.
+                base64_certificate_installation_res = (
+                    await self.comm_session.evse_controller.get_15118_ev_certificate(
+                        base64_certificate_install_req, Namespace.ISO_V2_MSG_DEF
+                    )
+                )
+                certificate_installation_res: Base64 = Base64(
+                    payload=base64_certificate_installation_res,
+                    payload_type=CertificateInstallationRes.__name__,
+                )
+            else:
+                (
+                    certificate_installation_res,
+                    signature,
+                ) = self.generate_certificate_installation_res()
+        except Exception as e:
+            logger.error(f"Error building CertificateInstallationRes: {e}")
+            self.stop_state_machine(
+                str(e),
+                message,
+                ResponseCode.FAILED_NO_CERTIFICATE_AVAILABLE,
+            )
+            return
+
+        self.create_next_message(
+            PaymentDetails,
+            certificate_installation_res,
+            Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
+            Namespace.ISO_V2_MSG_DEF,
+            signature=signature,
         )
 
+    def validate_message_signature(self, message: V2GMessageV2) -> bool:
         # For the CertificateInstallation, the min. the SECC can do is
         # to verify the message signature, using the OEM provisioning
-        # certificate (public key)
+        # certificate (public key) - this is available in the cert installation req.
         # The chain of signatures, from the signature in the
         # CertificateInstallationReq's header all the way to the
         # self-signed OEM/V2G root certificate, can be verified if the
         # OEM Sub-CA and OEM/V2G root CA certificate are available.
-        if not verify_signature(
-            signature=msg.header.signature,
+
+        cert_install_req: CertificateInstallationReq = (
+            message.body.certificate_installation_req
+        )
+
+        sub_ca_certificates_oem = None
+        root_ca_certificate_oem = None
+        if (
+            os.path.exists(CertPath.OEM_SUB_CA2_DER)
+            and os.path.exists(CertPath.OEM_SUB_CA1_DER)
+            and os.path.exists(CertPath.OEM_ROOT_DER)
+        ):
+            sub_ca_certificates_oem = [
+                load_cert(CertPath.OEM_SUB_CA2_DER),
+                load_cert(CertPath.OEM_SUB_CA1_DER),
+            ]
+            root_ca_certificate_oem = CertPath.OEM_ROOT_DER
```

Comment:
`load_cert` already raises an exception if the certificates cant be fetched or is there an error

So, I would do the following:
```python
try:
    sub_ca_certificates_oem = [
        load_cert(CertPath.OEM_SUB_CA2_DER),
        load_cert(CertPath.OEM_SUB_CA1_DER),
    ]
    root_ca_certificate_oem = load_cert(CertPath.OEM_ROOT_DER)
except (FileNotFoundError, IOError):
    pass
```